google/resource_container_node_pool.go

```diff
@@ -155,11 +151,81 @@ var schemaNodePool = map[string]*schema.Schema{
 	},
 }
 
-func resourceContainerNodePoolCreate(d *schema.ResourceData, meta interface{}) error {
-	containerApiVersion := getContainerApiVersion(d, ContainerNodePoolBaseApiVersion, ContainerNodePoolVersionedFeatures)
-	config := meta.(*Config)
+func getNodePoolRegion(d TerraformResourceData, config *Config) (string, error) {
+	res, ok := d.GetOk("region")
```

Comment:
@danawillow I'm not too sure what you mean. Are you saying an error would occur when the node pool is nested?

Reply:
Sure, happy to elaborate. You can define a cluster with a node pool resource inside of it:

```
resource "google_container_cluster" "with_node_pool" {
	name = "cluster"
	zone = "us-central1-a"

	node_pool {
		name               = "np"
		initial_node_count = 2
	}
}
```

If you check out the code for `google_container_cluster`, you can see where it calls into some of the functions inside `resource_container_node_pool.go`. If you do a d.Get() from there, you're still doing it on the cluster resource, and so it'll use the cluster schema rather than the node pool schema. That's why a lot of the code inside of the node pool resource looks like `d.Get(prefix + "something")`: that lets the cluster and node pool resource share the code.